Content.Server/Explosion/EntitySystems/ExplosionSystem.Processing.cs

```diff
@@ -0,0 +1,647 @@
+        {
+            var ev = new GetExplosionResistanceEvent(id);
+            RaiseLocalEvent(uid, ev, false);
+
+            if (ev.Resistance == 0)
+            {
+                // no damage-dict multiplication required.
+                _damageableSystem.TryChangeDamage(uid, damage, ignoreResistances: true, damageable: damageable);
+            }
+            else if (ev.Resistance < 1)
+            {
+                _damageableSystem.TryChangeDamage(uid, damage * (1 - ev.Resistance), ignoreResistances: true, damageable: damageable);
+            }
+        }
+
+        // throw
+        if (xform != null
+            && !xform.Anchored 
+            && throwForce > 0
+            && !EntityManager.IsQueuedForDeletion(uid)
+            && physicsQuery.TryGetComponent(uid, out var physics)
+            && physics.BodyType == BodyType.Dynamic)
+        {
+            // TODO purge throw helpers and pass in physics component
+            uid.TryThrow(xform.WorldPosition - epicenter.Position, throwForce);
+        }
+
+        // TODO EXPLOSION puddle / flammable ignite?
+
+        // TODO EXPLOSION deaf/ear damage? other explosion effects?
+    }
+
+    /// <summary>
+    ///     Tries to damage floor tiles. Not to be confused with the function that damages entities intersecting the
+    ///     grid tile.
+    /// </summary>
+    public void DamageFloorTile(TileRef tileRef,
+        float intensity,
+        List<(Vector2i GridIndices, Tile Tile)> damagedTiles,
+        ExplosionPrototype type)
+    {
+        var tileDef = _tileDefinitionManager[tileRef.Tile.TypeId];
+
+        while (_robustRandom.Prob(type.TileBreakChance(intensity)))
+        {
+            intensity -= type.TileBreakRerollReduction;
+
+            if (tileDef is not ContentTileDefinition contentTileDef)
+                break;
+
+            // does this have a base-turf that we can break it down to?
+            if (contentTileDef.BaseTurfs.Count == 0)
+                break;
+
+            tileDef = _tileDefinitionManager[contentTileDef.BaseTurfs[^1]];
+        }
+
+        if (tileDef.TileId == tileRef.Tile.TypeId)
+            return;
+
+        damagedTiles.Add((tileRef.GridIndices, new Tile(tileDef.TileId)));
+    }
+}
+
+/// <summary>
+///     This is a data class that stores information about the area affected by an explosion, for processing by <see
+///     cref="ExplosionSystem"/>.
+/// </summary>
+/// <remarks>
+///     This is basically the output of <see cref="ExplosionSystem.GetExplosionTiles()"/>, but wrapped in an enumerator
+///     to iterate over the tiles, along with the ability to keep track of what entities have already been damaged by
+///     this explosion.
+/// </remarks>
+sealed class Explosion
+{
+    struct ExplosionData
+    {
+        public EntityLookupComponent Lookup;
+        public Dictionary<int, List<Vector2i>> TileLists;
+        public IMapGrid? MapGrid;
+    }
```

Comment:
does this correspond to one grid thats being exploded, or? could just use comments

Reply:
Either a grid, or space. Will add comments